Lib/asyncio/base_events.py

```diff
@@ -961,7 +961,10 @@ async def _connect_sock(self, exceptions, addr_info, local_addr_infos=None):
             sock = socket.socket(family=family, type=type_, proto=proto)
             sock.setblocking(False)
             if local_addr_infos is not None:
-                for _, _, _, _, laddr in local_addr_infos:
+                for lfamily, _, _, _, laddr in local_addr_infos:
+                    # skip local addresses of different family
+                    if lfamily != family:
+                        continue
```

Comment:
If somehow _all_ local addresses are of a different family, we end up taking the `else` clause on the `for` statement with `my_exceptions` still empty, and the `pop()` call will fail. I don't know how easy it is to test this case (you may have to call this method directly or use a mock) but it seems a legitimate case. I guess the `raise` below would have to make up some exception.

Reply:
Seems very unlikely but I fixed it and added a test for it.